handler.go

```diff
@@ -62,11 +68,16 @@ func (d *defaultHandler) AddHandler(function interface{}) error {
 	// Get event type
 	eventType := content.Type()
 
+	d.mut.Lock()
+
 	// Add it to the list of handlers
 	if _, ok := d.handlers[eventType]; !ok {
 		d.handlers[eventType] = make([]reflect.Value, 0, 1)
 	}
 	d.handlers[eventType] = append(d.handlers[eventType], val)
+
+	d.mut.Unlock()
```

Comment:
Please move the `Unlock` to under the `Lock`.

Reply:
Doing this would cause the lock to be locked while the log is being (color) rendered and written, which seems wasteful to me.

Reply:
Imo it wouldn't cost much perf if debug was not enabled. You're free to just mark this as resolved if you choose to however.

Reply:
That's a good point.